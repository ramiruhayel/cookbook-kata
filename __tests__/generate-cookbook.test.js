const generateCookbook = require("../lib/generate-cookbook.js");

describe("generate-cookbook", () => {
  const barfaliciousBrowniesRecipe = {
    name: "Barfalicious Brownies",
    ingredients: [
      { name: "sour milk", quantity: "3 cups" },
      { name: "pop rocks", quantity: "3 tsp" },
      { name: "milo", quantity: "1 cup" },
    ],
    instructions: [
      "Preheat oven to 180C",
      "Mix ingredients",
      "Pour into baking tray",
      "Cook for 25 mins",
      "Barf!",
    ],
  };

  const memberMuffinsRecipe = {
    name: "'Member Muffins?",
    ingredients: [
      { name: "milk", quantity: "1 cup" },
      { name: "sugar", quantity: "2 tbs" },
      { name: "flour", quantity: "2 cups" },
    ],
    instructions: [
      "Preheat oven to 220C",
      "Mix ingredients",
      "Pour into backing tray",
      "Cook for 15 mins",
      "Enjoy!",
    ],
  };
  it("generates an empty cookbook when there are no recipes", () => {
    expect(generateCookbook([])).toMatchSnapshot();
  });

  it.only("generates a cookbook given a single recipe", () => {
    expect(generateCookbook([memberMuffinsRecipe])).toMatchSnapshot();
  });
  it("generates a cookbook given a two recipes", () => {
    expect(
      generateCookbook([memberMuffinsRecipe, barfaliciousBrowniesRecipe])
    ).toMatchSnapshot();
  });
  it("generates a cookbook given 10 recipes", () => {
    expect(
      generateCookbook(
        Array(5)
          .fill(memberMuffinsRecipe)
          .concat(Array(5).fill(barfaliciousBrowniesRecipe))
      )
    ).toMatchSnapshot();
  });
});
