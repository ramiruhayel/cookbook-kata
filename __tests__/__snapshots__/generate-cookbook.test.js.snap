// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-cookbook generates a cookbook given 10 recipes 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\" />
    <title>My Team</title>
  </head>
  <body>
    <section>
      <h2>'Member Muffins?</h2>
      <h4>Ingredients:</h4>
      <ul>
        <li>milk <b>1 cup</b></li>
        <li>sugar <b>2 tbs</b></li>
        <li>flour <b>2 cups</b></li>
      </ul>
      <h4>Instructions:</h4>
      <ol>
        <li>Preheat oven to 220C</li>
        <li>Mix ingredients</li>
        <li>Pour into backing tray</li>
        <li>Cook for 15 mins</li>
        <li>Enjoy!</li>
      </ol>
    </section>
    <section>
      <h2>'Member Muffins?</h2>
      <h4>Ingredients:</h4>
      <ul>
        <li>milk <b>1 cup</b></li>
        <li>sugar <b>2 tbs</b></li>
        <li>flour <b>2 cups</b></li>
      </ul>
      <h4>Instructions:</h4>
      <ol>
        <li>Preheat oven to 220C</li>
        <li>Mix ingredients</li>
        <li>Pour into backing tray</li>
        <li>Cook for 15 mins</li>
        <li>Enjoy!</li>
      </ol>
    </section>
    <section>
      <h2>'Member Muffins?</h2>
      <h4>Ingredients:</h4>
      <ul>
        <li>milk <b>1 cup</b></li>
        <li>sugar <b>2 tbs</b></li>
        <li>flour <b>2 cups</b></li>
      </ul>
      <h4>Instructions:</h4>
      <ol>
        <li>Preheat oven to 220C</li>
        <li>Mix ingredients</li>
        <li>Pour into backing tray</li>
        <li>Cook for 15 mins</li>
        <li>Enjoy!</li>
      </ol>
    </section>
    <section>
      <h2>'Member Muffins?</h2>
      <h4>Ingredients:</h4>
      <ul>
        <li>milk <b>1 cup</b></li>
        <li>sugar <b>2 tbs</b></li>
        <li>flour <b>2 cups</b></li>
      </ul>
      <h4>Instructions:</h4>
      <ol>
        <li>Preheat oven to 220C</li>
        <li>Mix ingredients</li>
        <li>Pour into backing tray</li>
        <li>Cook for 15 mins</li>
        <li>Enjoy!</li>
      </ol>
    </section>
    <section>
      <h2>'Member Muffins?</h2>
      <h4>Ingredients:</h4>
      <ul>
        <li>milk <b>1 cup</b></li>
        <li>sugar <b>2 tbs</b></li>
        <li>flour <b>2 cups</b></li>
      </ul>
      <h4>Instructions:</h4>
      <ol>
        <li>Preheat oven to 220C</li>
        <li>Mix ingredients</li>
        <li>Pour into backing tray</li>
        <li>Cook for 15 mins</li>
        <li>Enjoy!</li>
      </ol>
    </section>
    <section>
      <h2>Barfalicious Brownies</h2>
      <h4>Ingredients:</h4>
      <ul>
        <li>sour milk <b>3 cups</b></li>
        <li>pop rocks <b>3 tsp</b></li>
        <li>milo <b>1 cup</b></li>
      </ul>
      <h4>Instructions:</h4>
      <ol>
        <li>Preheat oven to 180C</li>
        <li>Mix ingredients</li>
        <li>Pour into baking tray</li>
        <li>Cook for 25 mins</li>
        <li>Barf!</li>
      </ol>
    </section>
    <section>
      <h2>Barfalicious Brownies</h2>
      <h4>Ingredients:</h4>
      <ul>
        <li>sour milk <b>3 cups</b></li>
        <li>pop rocks <b>3 tsp</b></li>
        <li>milo <b>1 cup</b></li>
      </ul>
      <h4>Instructions:</h4>
      <ol>
        <li>Preheat oven to 180C</li>
        <li>Mix ingredients</li>
        <li>Pour into baking tray</li>
        <li>Cook for 25 mins</li>
        <li>Barf!</li>
      </ol>
    </section>
    <section>
      <h2>Barfalicious Brownies</h2>
      <h4>Ingredients:</h4>
      <ul>
        <li>sour milk <b>3 cups</b></li>
        <li>pop rocks <b>3 tsp</b></li>
        <li>milo <b>1 cup</b></li>
      </ul>
      <h4>Instructions:</h4>
      <ol>
        <li>Preheat oven to 180C</li>
        <li>Mix ingredients</li>
        <li>Pour into baking tray</li>
        <li>Cook for 25 mins</li>
        <li>Barf!</li>
      </ol>
    </section>
    <section>
      <h2>Barfalicious Brownies</h2>
      <h4>Ingredients:</h4>
      <ul>
        <li>sour milk <b>3 cups</b></li>
        <li>pop rocks <b>3 tsp</b></li>
        <li>milo <b>1 cup</b></li>
      </ul>
      <h4>Instructions:</h4>
      <ol>
        <li>Preheat oven to 180C</li>
        <li>Mix ingredients</li>
        <li>Pour into baking tray</li>
        <li>Cook for 25 mins</li>
        <li>Barf!</li>
      </ol>
    </section>
    <section>
      <h2>Barfalicious Brownies</h2>
      <h4>Ingredients:</h4>
      <ul>
        <li>sour milk <b>3 cups</b></li>
        <li>pop rocks <b>3 tsp</b></li>
        <li>milo <b>1 cup</b></li>
      </ul>
      <h4>Instructions:</h4>
      <ol>
        <li>Preheat oven to 180C</li>
        <li>Mix ingredients</li>
        <li>Pour into baking tray</li>
        <li>Cook for 25 mins</li>
        <li>Barf!</li>
      </ol>
    </section>
  </body>
</html>
"
`;

exports[`generate-cookbook generates a cookbook given a single recipe 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\" />
    <title>My Team</title>
  </head>
  <body>
  </body>
</html>
"
`;

exports[`generate-cookbook generates a cookbook given a two recipes 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\" />
    <title>My Team</title>
  </head>
  <body>
    <section>
      <h2>'Member Muffins?</h2>
      <h4>Ingredients:</h4>
      <ul>
        <li>milk <b>1 cup</b></li>
        <li>sugar <b>2 tbs</b></li>
        <li>flour <b>2 cups</b></li>
      </ul>
      <h4>Instructions:</h4>
      <ol>
        <li>Preheat oven to 220C</li>
        <li>Mix ingredients</li>
        <li>Pour into backing tray</li>
        <li>Cook for 15 mins</li>
        <li>Enjoy!</li>
      </ol>
    </section>
    <section>
      <h2>Barfalicious Brownies</h2>
      <h4>Ingredients:</h4>
      <ul>
        <li>sour milk <b>3 cups</b></li>
        <li>pop rocks <b>3 tsp</b></li>
        <li>milo <b>1 cup</b></li>
      </ul>
      <h4>Instructions:</h4>
      <ol>
        <li>Preheat oven to 180C</li>
        <li>Mix ingredients</li>
        <li>Pour into baking tray</li>
        <li>Cook for 25 mins</li>
        <li>Barf!</li>
      </ol>
    </section>
  </body>
</html>
"
`;

exports[`generate-cookbook generates an empty cookbook when there are no recipes 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\" />
    <title>My Team</title>
  </head>
  <body></body>
</html>
"
`;
